build.psm1

```diff
@@ -835,7 +836,14 @@ function Test-PSPesterResults
     if ([int]$x.'test-results'.failures -gt 0)
     {
         logerror "TEST FAILURES"
-        foreach ( $testfail in $x.SelectNodes('.//test-case[@result = "Failure"]'))
+        # switch between methods, SelectNode is not available on dotnet core
+        if ( "System.Xml.XmlDocumentXPathExtensions" -as "Type" ) {
```

Comment:
As a general rule, you should prefer `-as [Type]` instead of `-as "Type"` because the former resolves the type just once while the later resolves it every time it is executed.